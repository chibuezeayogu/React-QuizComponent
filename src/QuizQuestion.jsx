import React, { Component } from 'react'
import { shape, func } from 'prop-types';

import QuizQuestionButton from './QuizQuestionButton'

/**
 * 
 * @class QuizQuestion
 * @extends Component
 */
class QuizQuestion extends Component {
  constructor(props) {
    super(props)

    this.state = { incorrectAnswer: null };
  }

  /**
   * 
   * @method handleClick
   * 
   * @memberof QuizEnd
   * 
   * @returns { void }
   */
  handleClick = (button_text) => {
    if (button_text === this.props.quiz_question.answer) {
      this.setState({ incorrectAnswer: false });
      this.props.showNextQuestionHandler()
    } else {
      this.setState({ incorrectAnswer: true });
    }
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberof QuizQuestion
   *
   * @returns { JSX }
   *
   */
  render() {
    const { quiz_question } = this.props;
    const { incorrectAnswer } = this.state;
    return (
      <>
      <div className="question">
        {quiz_question.instruction_text}
      </div>
      <div className="answers-container">
        {quiz_question.answer_options.map((answer, index) => (
          <QuizQuestionButton 
            button_text={answer} 
            key={index} 
            clickHandler={this.handleClick} 
          />
        ))}
      </div>
        {incorrectAnswer ?
          <p className='error'>Sorry, that's not right</p> : 
          null
        }
      </>                
    )
  }
}

QuizQuestion.propTypes = {
  quiz_question: shape({}),
  showNextQuestionHandler: func
};

export default QuizQuestion;
