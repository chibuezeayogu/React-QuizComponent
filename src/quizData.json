{
  "quizQuestions": [
    {
      "id": 1,
      "instructionText": "How many continents are there on Planet Earth?",
      "answerOptions": ["5", "6", "7", "8"],
      "answer": "7"
    },
    {
      "id": 2,
      "instructionText": "Greek civilization was flourished in the year?",
      "answerOptions": ["1000 BC", "900 BC", "700 BC", "800 BC"],
      "answer": "800 BC"
    },
    {
      "id": 3,
      "instructionText": "What is the largest continent on earth?",
      "answerOptions": ["Asia", "Africa", "North America", "South America"],
      "answer": "Asia"
    },
    {
      "id": 4,
      "instructionText": "Rome was founded in the year?",
      "answerOptions": ["754 BC", "755 BC", "753 BC", "756 BC"],
      "answer": "753 BC"
    },
    {
      "id": 5,
      "instructionText": " Who was first to reach the South Pole?",
      "answerOptions": ["Capt. Austin", "Capt. White", "Capt. Peter", "Capt. Amundsen"],
      "answer": "Capt. Amundsen"
    },
    {
      "id": 6,
      "instructionText": "Buckingham Palace became the official royal residence during the reign of which British monarch?",
      "answerOptions": ["Queen Elizbeth", "Queen Veronica", "Queen Victoria", "Queen Elizbeth II"],
      "answer": "Queen Victoria"
    },
    {
      "id": 7,
      "instructionText": "In which year Bangladesh was formed as Separate Country?",
      "answerOptions": ["1973", "1971", "1197", "1974"],
      "answer": "1971"
    },
    {
      "id": 8,
      "instructionText": "The Great Pyramid of Giza was built for which Egyptian ruler?",
      "answerOptions": ["Narmer", "Menes", "Cheops", "Khufu"],
      "answer": "Cheops"
    },
    {
      "id": 9,
      "instructionText": "World War I commenced in?",
      "answerOptions": ["1919", "1914", "1913", "1917"],
      "answer": "1914"
    },
    {
      "id": 10,
      "instructionText": "Who is considered the \"Father of the Constitution\"?",
      "answerOptions": ["James Madison", "Peter Madison", "Paul Madison", "Solomon Madison"],
      "answer": "James Madison"
    }
  ]
}