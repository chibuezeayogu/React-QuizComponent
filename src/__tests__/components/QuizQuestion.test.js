import React from 'react';
import { shallow } from 'enzyme';

import QuizQuestion from '../../QuizQuestion';
import quizData from '../../quizData.json';

describe('QuizQuestion Component', () => {
  const setup = () => {
    const props = {
      showNextQuestionHandler: jest.fn(),
      quizQuestion: {
        answerOptions: []
      }
    }

    return shallow(<QuizQuestion {...props} />);
  };

  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleResetClick method with right answer', () => {
    const handleSpy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().handleClick();
    expect(handleSpy).toHaveBeenCalled();
  });

  it('should call handleClick method with wrong answer', () => {
    const handleSpy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().handleClick("0");
    expect(handleSpy).toHaveBeenCalled();
  });
});
