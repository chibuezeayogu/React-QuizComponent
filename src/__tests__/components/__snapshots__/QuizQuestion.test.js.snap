// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizQuestion Component should render correctly 1`] = `
<React.Fragment>
  <div
    className="question"
  />
  <div
    className="answers-container"
  />
</React.Fragment>
`;
